File: CeliTax/TriangleView.m
//
// triangleView.m
// CeliTax
//
// Created by Leon Chen on 2015-06-10.
// Copyright (c) 2015 CraveNSave. All rights reserved.
//

#import "TriangleView.h"

@implementation TriangleView

- (void) baseInit
{
    self.backgroundColor = [UIColor clearColor];
    
    [self setGreenArrowDown];
}

- (instancetype) initWithFrame: (CGRect) frame
{
    self = [super initWithFrame: frame];

    if (self)
    {
        [self baseInit];
    }

    return self;
}

- (instancetype) initWithCoder: (NSCoder *) aDecoder
{
    self = [super initWithCoder: aDecoder];

    if (self)
    {
        [self baseInit];
    }

    return self;
}

- (void) setGreenArrowUp
{
    self.image = [UIImage imageNamed: @"greenTrianglePointUp"];
}

- (void) setGreenArrowDown
{
    self.image = [UIImage imageNamed: @"greenTrianglePointDown"];
}

@end

File: CeliTax/TriangleView.swift
//
//  TriangleView.swift
//  CeliTax
//
//  Created by Leon Chen on 2015-11-28.
//  Copyright © 2015 CraveNSave. All rights reserved.
//

import Foundation

@objc
class TriangleView : UIImageView
{
    func baseInit()
    {
        self.backgroundColor = UIColor.clearColor()
    
        self.setGreenArrowDown()
    }
    
    override init (frame : CGRect)
    {
        super.init(frame : frame)
        self.baseInit()
    }
    
    required init?(coder aDecoder: NSCoder)
    {
        super.init(coder: aDecoder)
        self.baseInit()
    }
    
    func setGreenArrowUp()
    {
        self.image = UIImage.init(named: "greenTrianglePointUp")
    }
    
    func setGreenArrowDown()
    {
        self.image = UIImage.init(named: "greenTrianglePointDown")
    }

}